source/extensions/http/stateful_session/header/header.h

```diff
@@ -38,25 +38,15 @@ class HeaderBasedSessionStateFactory : public Envoy::Http::SessionStateFactory {
 
   HeaderBasedSessionStateFactory(const HeaderBasedSessionStateProto& config);
 
-  Envoy::Http::SessionStatePtr create(const Envoy::Http::RequestHeaderMap& headers) const override {
+  Envoy::Http::SessionStatePtr create(Envoy::Http::RequestHeaderMap& headers) const override {
     return std::make_unique<SessionStateImpl>(parseAddress(headers), *this);
   }
 
 private:
-  absl::optional<std::string> parseAddress(const Envoy::Http::RequestHeaderMap& headers) const {
-    auto hdr = headers.get(Envoy::Http::LowerCaseString(name_));
-    if (hdr.empty()) {
-      return absl::nullopt;
-    }
-
-    auto header_value = hdr[0]->value().getStringView();
-    std::string address = Envoy::Base64::decode(header_value);
-    return !address.empty() ? absl::make_optional(std::move(address)) : absl::nullopt;
-  }
-
-  const Envoy::Http::LowerCaseString& getHeaderName() const { return name_; }
+  absl::optional<std::string> parseAddress(Envoy::Http::RequestHeaderMap& headers) const;
 
   const Envoy::Http::LowerCaseString name_;
+  const HeaderBasedSessionStateProto::Mode mode_{};
```

Comment:
Maybe it should be initialized to default `OVERRIDE` mode?

Reply:
The `{}` will initialize the `mode_` to `0`, which means `OVERRIDE`. And the will always initialize the mode_ at the constructor initialization list with `proto_config.mode()`.

Reply:
Got it. Thanks.